UnitTests/Application/ApplicationTests.cs

```diff
@@ -108,6 +110,7 @@ public void Init_Begin_End_Cleans_Up ()
         Assert.NotNull (Application.MainLoop);
         Assert.NotNull (Application.Driver);
 
+        topLevel.Dispose ();
```

Comment:
Somwhere we need a unit test that proves `Appliciation._latestClosedRunStateToplevel` was reset correctly.

Reply:
It's working correctly. On all unit tests that use `Application.Shutdown` I had to precede them by the object.Dispose(), otherwise it'll fail the assertion.